ui/lib/pki/addon/routes/certificates/index.js

```diff
@@ -14,23 +14,35 @@ export default class PkiCertificatesIndexRoute extends Route {
   @service store;
   @service secretMountPath;
 
-  async fetchCertificates() {
+  queryParams = {
+    page: {
+      refreshModel: true,
+    },
+  };
+
+  async fetchCertificates(params) {
     try {
-      return await this.store.query('pki/certificate/base', { backend: this.secretMountPath.currentPath });
+      const page = Number(params.page) || 1;
+      return await this.store.lazyPaginatedQuery('pki/certificate/base', {
+        backend: this.secretMountPath.currentPath,
+        responsePath: 'data.keys',
+        page,
+        skipCache: page === 1,
```

Comment:
This allows the dataset to be refreshed whenever the list page is visited from a different route, or the user goes back to page 1